### PR TITLE
Isolate Exporter / DataItemToElementMapper

### PR DESCRIPTION
**Pull request by mwjames**

### CODE REVIEW

**Comment by mwjames** [comment]
Requires #965 to be merged first

File: src/Exporter/DataItemToElementMapper.php [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+	 * @return Element|null
+	 */
+	public function mapDataItemToElement( DataItem $dataItem ) {
+
+		if ( $this->dataItemToElementMappers === array() ) {
+				$this->initDataItemToElementMapper();
+		}
+
+		foreach ( $this->dataItemToElementMappers as $id => $dataItemToElementMapper ) {
+				if ( $id === $dataItem->getDIType() ) {
```
Comment (JeroenDeDauw) [inline src/Exporter/DataItemToElementMapper.php]: doubly indented

File: src/Exporter/DataItemToElementMapper.php [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+* @license GNU GPL v2+
+* @since 2.2
+*
+* @author mwjames
+* @author Markus Krötzsch
+*/
+class DataItemToElementMapper {
+
+	/**
+	 * @var array
```
Comment (JeroenDeDauw) [inline src/Exporter/DataItemToElementMapper.php]: `@var Closure[]`

Though actually you probably should be using `callable`, which includes `Closure`, though also allows for other things

File: src/Exporter/DataItemToElementMapper.php [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+					return $dataItemToElementMapper( $dataItem );
+				}
+		}
+
+		return null;
+	}
+
+	private function initDataItemToElementMapper() {
+
+		// Not used for now
```
Comment (JeroenDeDauw) [inline src/Exporter/DataItemToElementMapper.php]: ?

File: src/Exporter/DataItemToElementMapper.php [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+		} );
+
+		$this->registerDataItemToExpElementMapper( DataItem::TYPE_URI, function( $dataItem ) {
+			return new ExpResource(
+					$dataItem->getURI(),
+					$dataItem
+			);
+		} );
+
+		$this->registerDataItemToExpElementMapper( DataItem::TYPE_TIME, function( $dataItem ) use ( $lang ) {
```
Comment (JeroenDeDauw) [inline src/Exporter/DataItemToElementMapper.php]: For this one you are better off putting the implementation in a class, so you can split the function

**Comment by mwjames** [comment]
Only scrutinizer-ci failed.

**Pull request merged**